src/test/java/org/broadinstitute/consent/http/resources/DarCollectionResourceTest.java

```diff
@@ -0,0 +1,106 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.Date;
+import java.util.HashSet;
+
+import com.google.api.client.http.HttpStatusCodes;
+import org.apache.commons.lang3.RandomUtils;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+
+import org.broadinstitute.consent.http.enumeration.UserRoles;
+import org.broadinstitute.consent.http.models.AuthUser;
+import org.broadinstitute.consent.http.models.DataAccessRequest;
+import org.broadinstitute.consent.http.models.DataAccessRequestData;
+import org.broadinstitute.consent.http.models.dto.DatasetDTO;
+import org.broadinstitute.consent.http.models.DarCollection;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.UserRole;
+import org.broadinstitute.consent.http.service.DarCollectionService;
+import org.broadinstitute.consent.http.service.DatasetService;
+import org.broadinstitute.consent.http.service.UserService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class DarCollectionResourceTest {
+  private final AuthUser authUser = new AuthUser("test@test.com");
+  private final List<UserRole> researcherRole = Collections.singletonList(
+    new UserRole(UserRoles.RESEARCHER.getRoleId(), UserRoles.RESEARCHER.getRoleName())
+  );
+  private final User researcher = new User(1, authUser.getEmail(), "Display Name", new Date(), researcherRole, authUser.getEmail());
+
+  private DarCollectionResource resource;
+
+  @Mock private UserService userService;
+  @Mock private DatasetService datasetService;
+  @Mock private DarCollectionService darCollectionService;
+  @Mock private UriInfo info;
+  @Mock private UriBuilder builder;
+
+  private void initResource() {
+    resource = new DarCollectionResource(userService, darCollectionService, datasetService);
+  }
+
+  private DataAccessRequest mockDataAccessRequestWithDatasetIds() {
+    DataAccessRequest dar = new DataAccessRequest();
+    DataAccessRequestData data = new DataAccessRequestData();
+    data.setDatasetIds(Collections.singletonList(RandomUtils.nextInt(1, 100)));
+    dar.setData(data);
+    return dar;
+  }
+
+  private DarCollection mockDarCollection() {
+    DarCollection collection = new DarCollection();
+    collection.setDars(new ArrayList<DataAccessRequest>());
+    for(int i = 0; i < 3; i++) {
+      collection.getDars().add(mockDataAccessRequestWithDatasetIds());
+    }
+    return collection;
+  }
+
+  private Set<DatasetDTO> mockDatasetsForResearcherCollection() {
+    Set<DatasetDTO> datasets = new HashSet<DatasetDTO>();
+    for(int i = 1; i < 3; i++) {
```

Comment:
I find the various flavors of `IntStream.range()` much more legible than the old-style for loops.